pkg/store/gcworker/gc_worker.go

```diff
@@ -1236,21 +1232,119 @@ func (w *GCWorker) resolveLocks(
+	var allKeyspaces []*keyspacepb.KeyspaceMeta
+
+	if isNullKeyspace {
+		var err error
+		allKeyspaces, err = w.pdClient.GetAllKeyspaces(ctx, 0, loadAllKeyspacesForUnifiedGCTestsBatchSize)
+		if err != nil {
+			return err
+		}
+	}
+
+	// * If the current keyspace is not the null keyspace, then it must be a keyspace with keyspace level GC enabled.
+	//   in this case, resolve locks on the unbounded range, and the keyspace prefix will be automatically
+	//   attached.
+	// * If there are no keyspaces in the cluster at all, resolve locks for the unbounded whole key range.
+	if !isNullKeyspace || len(allKeyspaces) == 0 {
+		err := runner.RunOnRange(ctx, []byte(""), []byte(""))
+		if err != nil {
+			logutil.Logger(ctx).Warn("resolve locks failed", zap.String("category", "gc worker"),
+				zap.String("uuid", w.uuid),
+				zap.Uint64("txnSafePoint", txnSafePoint),
+				zap.Error(err))
+			return errors.Trace(err)
+		}
+
+		logutil.Logger(ctx).Info("finish resolve locks", zap.String("category", "gc worker"),
 			zap.String("uuid", w.uuid),
-			zap.Uint64("safePoint", safePoint),
-			zap.Error(err))
-		return errors.Trace(err)
+			zap.Uint64("txnSafePoint", txnSafePoint),
+			zap.Int("regions", runner.CompletedRegions()))
+		metrics.GCHistogram.WithLabelValues("resolve_locks").Observe(time.Since(startTime).Seconds())
+		return nil
 	}
 
-	logutil.Logger(ctx).Info("finish resolve locks", zap.String("category", "gc worker"),
-		zap.String("uuid", w.uuid),
-		zap.Uint64("safePoint", safePoint),
-		zap.Int("regions", runner.CompletedRegions()))
-	metrics.GCHistogram.WithLabelValues("resolve_locks").Observe(time.Since(startTime).Seconds())
+	// Otherwise, the null keyspace, which is the current keyspace, has the responsibility to resolve locks forr
+	// other keyspaces that are configured running unified GC, but skip keyspaces that use keyspace level GC.
+
+	isSuccessful := true
+
+	// First, resolve locks for the null keyspace (txn key range).
+	{
+		nullKeyspaceExcludePrefixes := tikv.CodecV1ExcludePrefixes()
+		nullKeyspaceKeyRanges := make([]tikvstore.KeyRange, 0, len(nullKeyspaceExcludePrefixes)+1)
+		nextStartKey := []byte("")
+
+		for _, prefix := range nullKeyspaceExcludePrefixes {
+			nullKeyspaceKeyRanges = append(nullKeyspaceKeyRanges, tikvstore.KeyRange{
+				StartKey: nextStartKey,
+				EndKey:   prefix,
+			})
+			nextStartKey = tikvstore.PrefixNextKey(prefix)
+		}
+
+		// If there are prefixes that has been processed but `nextStartKey` is set to empty, it means that there
+		// exist a prefix full of 0xff, causing the PrefixNextKey founds the global end. So no remaining range
+		// is needed to be added.
+		// Currently, there's no such prefix in use, but we handle it in code for strictness.
+		if !(len(nullKeyspaceExcludePrefixes) > 0 && len(nextStartKey) == 0) {
+			nullKeyspaceKeyRanges = append(nullKeyspaceKeyRanges, tikvstore.KeyRange{
+				StartKey: nextStartKey,
+				EndKey:   []byte(""),
+			})
+		}
```

Comment:
Seems to me this paragraph is to add a the final range of [succ(prefix), nil). But the comments are not obvious enough to me... At first sight it makes me think this `if` is an edge case that won't be triggered in current implementation.
Am I understanding it correctly?